Packs/rasterize/Integrations/rasterize/rasterize.py

```diff
@@ -362,55 +362,190 @@ def start_chrome_headless(chrome_port, chrome_binary=CHROME_EXE, user_options=""
+                       and not any(identifier in process for identifier in chrome_renderer_identifiers)]
+
+    if killall:
+        # fetch the pids of the processes
+        pids = [int(process.split()[1]) for process in process_in_list]
+    else:
+        # fetch the pid of the process. the list contain just one process with the given chrome_port
+        process_string_representation = process_in_list[0]
+        pids = [int(process_string_representation.split()[1])]
+
+    for pid in pids:
+        # for each pid, get the process by it PID and terminate it
+        process = psutil.Process(pid)
+        if process:
+            try:
+                demisto.debug(f'terminate_chrome, {process=}')
+                process.kill()
+            except Exception as e:
+                demisto.info(f"Exception when trying to kill chrome with {pid=}, {e}")
 
     demisto.debug('terminate_chrome, Finish')
 
 
-def ensure_chrome_running():  # pragma: no cover
+def chrome_manager() -> tuple[Any | None, str | None]:
+    """
+    Manages Chrome instances based on user-specified chrome options and integration instance ID.
+
+    This function performs the following steps:
+    1. Retrieves the instance ID of the integration and the Chrome options set by the user.
+    2. Checks if the instance ID has been used previously.
+        - If the instance ID is new, generates a new Chrome instance with the specified Chrome options.
+        - If the instance ID has been used:
+            - If the current Chrome options differ from the saved options for this instance ID,
+              it terminates the existing Chrome instance and generates a new one with the new options.
+            - If the current Chrome options match the saved options for this instance ID,
+              it reuses the existing Chrome instance.
 
-    # Check if we have a file with the port.
-    # If we have a file - Try to use it.
-    # If there's no file, or we cannot use it - Find a free port
-    browser = None
-    chrome_port = read_info_file(PORT_FILE_PATH)
-    if chrome_port:
-        browser = is_chrome_running_locally(chrome_port)
-        if browser:
-            return browser, chrome_port
-        else:
-            write_info_file(PORT_FILE_PATH, '')
+    Returns:
+        tuple[Any | None, int | None]: A tuple containing:
+            - The Browser or None if an error occurred.
+            - The chrome port or None if an error occurred.
+    """
+    # If instance_id or chrome_options are not set, assign 'None' to these variables.
+    # This way, when fetching the content from the file, if there was no instance_id or chrome_options before,
+    # it can compare between the fetched 'None' string and the 'None' that assigned.
+    instance_id = demisto.callingContext.get('context', {}).get('IntegrationInstanceID', 'None') or 'None'
+    chrome_options = demisto.params().get('chrome_options', 'None')
```

Comment:
```suggestion
    chrome_options = demisto.params().get('chrome_options') or 'None'
```
parameters always have a default value `None`, meaning - if `chrome_options` is not configured through the UI, its value will be `None` and you won't be able to set it via the 2nd argument of `get`.